gap_darwin.go

```diff
@@ -162,5 +162,9 @@ func (pd *peripheralDelegate) DidUpdateValueForCharacteristic(prph cbgo.Peripher
 		if char != nil && char.callback != nil {
 			go char.callback(chr.Value())
 		}
+
+		if char.readChan != nil {
+			char.readChan <- nil
```

Comment:
Reading a characteristic may result in an error, for example if the connection is dropped or a characteristic is not readable.
This code doesn't do any error handling and I think it should. For example, you could send a pair of (value, error) over the channel (in a new struct type for example).

Also, the doc comment above this function is now incorrect:

```go
// DidUpdateValueForCharacteristic is called when the characteristic for a Service
// for a Peripheral receives a notification with a new value.
```

Reply:
Changed.